pkg/chunk/grpc/grpc_client.go

```diff
@@ -0,0 +1,25 @@
+package grpc
+
+import (
+	"flag"
+
+	"google.golang.org/grpc"
+)
+
+// Config for a StorageClient
+type Config struct {
+	Address string `yaml:"address,omitempty"`
+}
+
+// RegisterFlags adds the flags required to config this to the given FlagSet
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	f.StringVar(&cfg.Address, "grpc-store.server-address", "", "Hostname or IP of gRPC instance.")
+}
+
+func connectToGrpcServer(serverAddress string) (GrpcStoreClient, *grpc.ClientConn, error) {
+	cc, err := grpc.Dial(serverAddress, grpc.WithInsecure())
```

Comment:
Given there's no health checking mechanism (ie. we're not using the client pool) we should setup the gRPC keepalive (we had few outages with etcd due to this). See `grpc.WithKeepaliveParams()` and the [etcd implementation](https://github.com/cortexproject/cortex/blob/c52331d3cbd4d8aa36106f0e2aeb3c1ecf8e0016/vendor/go.etcd.io/etcd/clientv3/client.go#L235-L240) as a reference.

Reply:
@pracucci Do you have any suggestions on these params. the keepalive check interval and the timeout duration when we don't have active rpcs?
I am thinking to put a 1 min interval to check is alive and 2 mins timeout interval to close the connection if no activity is noticed.
```
	params := keepalive.ClientParameters{
		Time:                time.Minute*1,
		Timeout:           time.Minute*2,
		PermitWithoutStream: true,
	}
```

Reply:
I would be even more aggressive, like: 
- Time: 20s
- Timeout: 10s
- PermitWithoutStream: true